include/mfmg/common/amge.templates.hpp

```diff
@@ -179,13 +179,16 @@ void AMGe<dim, VectorType>::build_agglomerate_triangulation(
   std::vector<typename dealii::DoFHandler<dim>::active_cell_iterator>
       agglomerate;
   auto cell = _dof_handler.begin_active();
-  std::advance(cell, cell_index[0]);
-  agglomerate.push_back(cell);
-
-  for (unsigned int i = 1; i < cell_index.size(); ++i)
+  if (cell_index.size() > 0)
   {
-    std::advance(cell, cell_index[i] - cell_index[i - 1]);
+    std::advance(cell, cell_index[0]);
     agglomerate.push_back(cell);
```

Comment:
From this diff it seems that `agglomerate` is always going to be of size `cell_index.size()`. Why not resize it first and avoid `push_back`?

Reply:
If we are concerned about performance here, let's use `reserve` instead.

Reply:
Just reserve enough space.

Reply:
`reserve` + `push_back` is still slower than `[]`... just saying

Reply:
Because of checking whether new size exceed capacity?

Reply:
push_back:
```c++
      void                                                                                                                                                                  
      push_back(const value_type& __x)                                                                                                                                      
      {                                                                                                                                                                     
    if (this->_M_impl._M_finish != this->_M_impl._M_end_of_storage)                                                                                                         
      {                                                                                                                                                                     
        _Alloc_traits::construct(this->_M_impl, this->_M_impl._M_finish,                                                                                                    
                     __x);                                                                                                                                                  
        ++this->_M_impl._M_finish;                                                                                                                                          
      }                                                                                                                                                                     
    else                                                                                                                                                                    
      _M_realloc_insert(end(), __x);                                                                                                                                        
      }   
```

operator[]:
```c++      
      reference                                                                                                                                                             
      operator[](size_type __n) _GLIBCXX_NOEXCEPT                                                                                                                           
      {                                                                                                                                                                     
    __glibcxx_requires_subscript(__n);                                                                                                                                      
    return *(this->_M_impl._M_start + __n);                                                                                                                                 
      }  

```

Reply:
It's hard to say if initializing the elements twice is worth the faster access or not...